vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt

```diff
@@ -476,22 +476,24 @@ class RoomDetailViewModel @AssistedInject constructor(
      * Convert a send mode to a draft and save the draft
      */
     private fun handleSaveDraft(action: RoomDetailAction.SaveDraft) = withState {
-        when {
-            it.sendMode is SendMode.REGULAR && !it.sendMode.fromSharing -> {
-                setState { copy(sendMode = it.sendMode.copy(action.draft)) }
-                room.saveDraft(UserDraft.REGULAR(action.draft), NoOpMatrixCallback())
-            }
-            it.sendMode is SendMode.REPLY                               -> {
-                setState { copy(sendMode = it.sendMode.copy(text = action.draft)) }
-                room.saveDraft(UserDraft.REPLY(it.sendMode.timelineEvent.root.eventId!!, action.draft), NoOpMatrixCallback())
-            }
-            it.sendMode is SendMode.QUOTE                               -> {
-                setState { copy(sendMode = it.sendMode.copy(text = action.draft)) }
-                room.saveDraft(UserDraft.QUOTE(it.sendMode.timelineEvent.root.eventId!!, action.draft), NoOpMatrixCallback())
-            }
-            it.sendMode is SendMode.EDIT                                -> {
-                setState { copy(sendMode = it.sendMode.copy(text = action.draft)) }
-                room.saveDraft(UserDraft.EDIT(it.sendMode.timelineEvent.root.eventId!!, action.draft), NoOpMatrixCallback())
+        viewModelScope.launch {
```

Comment:
We call this action when the Fragment is paused, so there is maybe a risk that the action is cancelled, because the `vieWModelScope` will be cancelled (I think). Have you tested that change on a device?

Reply:
`viewModelScope` is cancelled when the `ViewModel` is cleared, which happens when the owning `Activity`/`Fragment` is destroyed (so it will survive rotations and other config changes too).
Does this need to outlive the fragment's destruction?

To inspire confidence, I've linked docs that outline the behaviour.
- [ViewModelScope](https://developer.android.com/topic/libraries/architecture/coroutines#viewmodelscope)
- [ViewModel](https://developer.android.com/topic/libraries/architecture/viewmodel#implement)

I have not tested this on a device.

Reply:
> Does this need to outlive the fragment's destruction?

Yes, I think so, because we save the draft when the user leaves, and the RoomDetailActivity will be destroyed

Reply:
Oh okay. I'll probably just use `withContext(NonCancellable)` for it but I'll give it some thought first.

Reply:
Yes NonCancellable is the way to go I think 👍

Reply:
Ended up putting it in the `launch` directly, to ensure it doesn't get cancelled between `launch` and `withContext`.

Reply:
Thanks, LGTM now :)